src/main/java/donnafin/logic/parser/ParserUtil.java

```diff
@@ -240,25 +240,30 @@ public static Money parseMoney(String money) throws ParseException {
         String trimmedCommission = money.trim();
 
         // Handling Default currency $XYZ or $XYZ.AB
-        final String regexDollarCents = "^-?\\s*\\$\\s*\\d+(\\.\\d{2})?$";
+        final String regexDollarCents = "^\\s*\\$\\s*\\d+(\\.\\d{2})?$";
         final String dollarCentsPrefix = "$";
 
-        if (trimmedCommission.matches(regexDollarCents)) {
-            String decimalString = trimmedCommission.replace(dollarCentsPrefix, "").replace(" ", "");
-            if (!decimalString.contains(".")) {
-                decimalString += ".00";
-            }
-            long value;
-            try {
-                value = Long.parseLong(decimalString.replace(".", ""));
-            } catch (NumberFormatException e) {
-                throw new ParseException(
-                        String.format("Input string '%s' exceeds maximum monetary value.", trimmedCommission));
-            }
+        if (!trimmedCommission.matches(regexDollarCents)) {
+            throw new ParseException(
+                    String.format(
+                            "Input string '%s' does not match monetary value format. %s",
+                            trimmedCommission,
+                            Money.MESSAGE_CONSTRAINTS)
+            );
+        }
+
+        String decimalString = trimmedCommission.replace(dollarCentsPrefix, "").replace(" ", "");
+        if (!decimalString.contains(".")) {
+            decimalString += ".00";
+        }
+        try {
+            long value = Long.parseLong(decimalString.replace(".", ""));
             return new Money(value);
-        } else {
+        } catch (Money.MoneyException e) {
+            throw new ParseException(e.getMessage());
+        } catch (NumberFormatException e) {
             throw new ParseException(
-                    String.format("Input string '%s' does not match any monetary value format.", trimmedCommission));
+                    String.format("Input string '%s' exceeds maximum monetary value.", trimmedCommission));
```

Comment:
Should we leave a small explanation why it cannot be exceeded. maybe sth like SGD max can't be .....

Reply:
I've tried, take a look!